src/test/java/org/jmxtrans/agent/zabbix/TcpByteLineServer.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2010-2016 the original author or authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+package org.jmxtrans.agent.zabbix;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.jmxtrans.agent.GraphitePlainTextTcpOutputWriter;
+import org.junit.rules.ExternalResource;
+
+/**
+ * A server that listens on a TCP port and remembers received plain text lines.
+ * Suitable for testing output writers that write plain text lines such as {@link GraphitePlainTextTcpOutputWriter}.
+ * Can be used as a JUnit rule for automatic start/stop:
+ * 
+ * <pre>{@code @Rule
+ * public TcpLineServer server = new TcpLineServer();
+ * }</pre>
+ * 
+ * @author Kristoffer Erlandsson
```

Comment:
Good catch. That code was inspired by the Graphite testing code. I made the change.